rskj-core/src/test/java/co/rsk/peg/bitcoin/BitcoinUtilsTest.java

```diff
@@ -0,0 +1,262 @@
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+class BitcoinUtilsTest {
+    private static final BridgeConstants bridgeMainnetConstants = BridgeMainNetConstants.getInstance();
+    private static final NetworkParameters btcMainnetParams = bridgeMainnetConstants.getBtcParams();
+    private static final int FIRST_OUTPUT_INDEX = 0;
+    private static final int FIRST_INPUT_INDEX = 0;
+
+    List<BtcECKey> signers;
+    List<BtcECKey> pubKeys;
+    Address destinationAddress;
+
+    @BeforeEach
+    void init() {
+        signers = BitcoinTestUtils.getBtcEcKeysFromSeeds(
+            new String[]{"fa01", "fa02", "fa03"}, true
+        );
+        pubKeys = signers.stream().map(BtcECKey::getPubKeyPoint).map(BtcECKey::fromPublicOnly)
+            .collect(Collectors.toList());
+
+        destinationAddress = BitcoinTestUtils.createP2PKHAddress(btcMainnetParams, "destinationAddress");
+    }
+
+    @Test
+    void test_getFirstInputSigHash_from_multisig() {
+        // Arrange
+        int totalSigners = signers.size();
+        int requiredSignatures = totalSigners / 2 + 1;
+
+        Script redeemScript = ScriptBuilder.createRedeemScript(requiredSignatures, signers);
+
+        Address destinationAddress = BitcoinTestUtils.createP2PKHAddress(btcMainnetParams, "destinationAddress");
+
+        BtcTransaction btcTx = new BtcTransaction(btcMainnetParams);
+        btcTx.addInput(BitcoinTestUtils.createHash(1), FIRST_OUTPUT_INDEX, new Script(new byte[]{}));
+        btcTx.addOutput(Coin.COIN, destinationAddress);
+
+        RedeemData redeemData = RedeemData.of(pubKeys, redeemScript);
+        Script p2SHScript = ScriptBuilder.createP2SHOutputScript(requiredSignatures, pubKeys);
+        Script inputScript = p2SHScript.createEmptyInputScript(redeemData.keys.get(0), redeemData.redeemScript);
+        btcTx.getInput(FIRST_INPUT_INDEX).setScriptSig(inputScript);
+
+        Sha256Hash expectedSigHash = btcTx.hashForSignature(
+            FIRST_INPUT_INDEX,
+            redeemScript,
+            BtcTransaction.SigHash.ALL,
+            false
+        );
+
+        // Act
+        Optional<Sha256Hash> sigHash = BitcoinUtils.getFirstInputSigHash(btcTx);
+
+        // Assert
+        Assertions.assertTrue(sigHash.isPresent());
+        Assertions.assertEquals(expectedSigHash, sigHash.get());
+    }
+
+    @Test
+    void test_getFirstInputSigHash_from_fed() {
+        // Arrange
+        Federation federation = new Federation(
+            FederationTestUtils.getFederationMembersWithBtcKeys(signers),
+            Instant.ofEpochMilli(1000L),
+            0L,
+            btcMainnetParams
+        );
+
+        BtcTransaction btcTx = new BtcTransaction(btcMainnetParams);
+        btcTx.addInput(BitcoinTestUtils.createHash(1), FIRST_OUTPUT_INDEX, new Script(new byte[]{}));
+        btcTx.addOutput(Coin.COIN, destinationAddress);
+
+        Script fedInputScript = PegTestUtils.createBaseInputScriptThatSpendsFromTheFederation(federation);
```

Comment:
It's the same logic as the one used in the test above, no need to call this method from PegTestUtils. Just use ScriptBuilder

Reply:
Done